src/NuGet.Clients/PackageManagement.VisualStudio/SolutionRestore/SolutionRestoreJob.cs

```diff
@@ -357,9 +372,9 @@ private void DisplayOptOutMessage()
                 var packageIdentity = args.Package;
                 Interlocked.Increment(ref _currentCount);
 
-                ThreadHelper.JoinableTaskFactory.RunAsync(() =>
+                _logger.Do((_, progress) => 
                 {
-                    return RestoreOperationProgressUI.Current.ReportProgressAsync(
+                    progress.ReportProgress(
```

Comment:
It can be possible if outer calling method is changed in future. Will add a null-check here just in case.